### PR TITLE
fish: minor fixes

### PR DESCRIPTION
**Pull request by cole-h**
Just a few things I'd like to get in for the fish plugins PR.

* calls `escapeShellArg` on abbrs and aliases in order to allow apostrophes in the definition (some of my abbrs contain these)
* fixes sourcing of fish files in `conf.d` -- the quoting was the problem
* prepends the fenv function dir instead of concatenating the original array with that function dir (looks better IMO, but functionally equivalent -- feel free to drop that commit)

### CODE REVIEW

**Comment by ryanorendorff** [comment]
Looks good to me. I'm a bit rusty on fish; do the quotes also need to be removed from the `test` statements made before `source` is called?

Thanks for making the edits; I'm excited to see plugins supported by home manager!

**Comment by Jomik** [comment]
I have the same question as Ryan, other than that it looks good to me. Thank you! :)

**Comment by cole-h** [comment]
TL;DR: No, because there are no globs/wildcards happening in those test statements. It might make stylistic sense to either be all or nothing -- and thus nothing, since it won't function properly when `*.fish` is quoted, as we've established -- though. I can do that, just let me know :)

---

No they do not. The only reason the quotes need to be removed there is because they basically "escape" the `*` glob -- so it tries to source a file with name `~/.config/fish/conf.d/*.fish` instead of sourcing every file in `~/.config/fish/conf.d/` that has the extension `.fish`.

You can easily verify this by running the following:

```
set -l plg ~/.config/fish

if test -d $plg"/.conf.d"
  echo "conf.d exists!"
  ls $plg"/conf.d/*.fish" # `ls: cannot access 'fish/conf.d/*.fish': No such file or directory`
  ls $plg/conf.d/*.fish # fish/conf.d/plugin-nix-env.fish.fish, fish/conf.d/plugin-pure.fish, fish/conf.d/plugin-z.fish
end

if test -d $plg/.conf.d
  echo "conf.d exists!"
  ls $plg"/conf.d/*.fish" # `ls: cannot access 'fish/conf.d/*.fish': No such file or directory`
  ls $plg/conf.d/*.fish # fish/conf.d/plugin-nix-env.fish.fish, fish/conf.d/plugin-pure.fish, fish/conf.d/plugin-z.fish
end
```

**Comment by Jomik** [comment]
For consistency I feel like it would be better to remove them.

**Review by ryanorendorff — APPROVED** [review]
I say we merge it! :-)

**Comment by Jomik** [comment]
Awesome, thanks :)
